examples/launch_custom_benchmark_remotely.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""
+This example show how to launch a tuning job that will orchestrated on Sagemaker rather than on your local machine.
+"""
+
+import logging
+from pathlib import Path
+
+from benchmarking.commons.launch_remote_sagemaker import launch_remote
```

Comment:
You need to use launch_remote_X and hpo_main_X when you want to run with backend X in ["local", "sagemaker"].

What you do here, either does not work, or only works by accident.

Reply:
I see your point with using the specific launch_remote_X options, changed to use the appropriate function to be correct. However, I believe the code works (as it launches the appropriate benchmarks).

The code in  launch_remote_local and launch_remote_remote are very similar with less than 20% code line difference and most of them related to specifying the AWS region. The differences between the two functions do not have much impact when using the custom entrypoint from `launch_custom_benchmark.py` so I wouldn't say it's an accident exactly. I think this is actually a good argument to combine the two functions together to avoid duplication. I will probably try to merge them during the benchmarking entrypoint changes.

On a second point: hpo_main_X are used in `launch_custom_benchmark.py` which is the entrypoint for this benchmark.

Reply:
Then, let us do your first change first. I am not OK with examples that go against how I think this stuff should be used. I also see no reason of doing it the way you try to do here.

We can always unify common code, I like that. I did that to quite some extent already. However, running experiments with SM backend is very different from local backend, and blurring the distinction is not so useful, because it is really just there.